File: Java/Reverse Integer.java
// 1ms, 100.00%; 36.8MB, 5.55%
class Solution {
    public int reverse(int x) {
        int res = 0;
        while (x != 0) {
            int tmp = res * 10 + x % 10;
            if (tmp / 10 != res) {
                return 0;
            }
            res = tmp;
            x /= 10;
        }
        return res;
    }
}

File: Go/Reverse Integer.go
// 0ms, 100.00%; 2.2MB, 80.00%
func reverse(x int) int {
	res := 0
	for x != 0 {
		res = res*10 + x%10
		x /= 10
	}
	if res < -2147483648 || res > 2147483647 {
		res = 0
	}
	return res
}
